python/cugraph/community/subgraph_extraction_wrapper.pyx

```diff
@@ -32,52 +29,95 @@ import numpy as np
+    cdef GraphCOO[int,int,double] in_graph_double
+    cdef GraphCOO[int,int,float]  out_graph_float
+    cdef GraphCOO[int,int,double] out_graph_double
+
+    cdef uintptr_t c_src = src.__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_dst = dst.__cuda_array_interface__['data'][0]
+    cdef uintptr_t c_weights = <uintptr_t> NULL
+
+    if weights is not None:
+        c_weights = weights.__cuda_array_interface__['data'][0]
+    
+    if input_graph.renumbered:
         renumber_series = cudf.Series(input_graph.edgelist.renumber_map.index,
                                       index=input_graph.edgelist.renumber_map, dtype=np.int32)
         vertices_renumbered = renumber_series.loc[vertices]
-        vert_col = get_gdf_column_view(vertices_renumbered)
     else:
-        vert_col = get_gdf_column_view(vertices)
-
-    extract_subgraph_vertex_nvgraph(g, &vert_col, rg)
-
-    if rg.edgeList is not NULL:
-        df = cudf.DataFrame()
-        df['src'], df['dst'], vals = graph_wrapper.get_edge_list(rGraph)
-        if vals is not None:
-            df['val'] = vals
-            subgraph.from_cudf_edgelist(df, source='src', destination='dst', edge_attr='val')
-        else:
-            subgraph.from_cudf_edgelist(df, source='src', destination='dst')
-        if input_graph.edgelist is not None:
-            subgraph.renumbered = input_graph.renumbered
-            subgraph.edgelist.renumber_map = input_graph.edgelist.renumber_map
-    if rg.adjList is not NULL:
-        off, ind, vals = graph_wrapper.get_adj_list(rGraph)
-        subgraph.from_cudf_adjlist(off, ind, vals)
-    if rg.transposedAdjList is not NULL:
-        off, ind, vals = graph_wrapper.get_transposed_adj_list(rGraph)
-        subgraph.transposedadjlist = subgraph.transposedAdjList(off, ind, vals)
+        vertices_renumbered = vertices
+
+    cdef uintptr_t c_vertices = vertices_renumbered.__cuda_array_interface__['data'][0]
+
+    num_verts = input_graph.number_of_vertices()
+    num_edges = len(src)
+    num_input_vertices = len(vertices)
+
+    df = cudf.DataFrame()
+
+    if use_float:
+        in_graph_float = GraphCOO[int,int,float](<int*>c_src, <int*>c_dst, <float*>c_weights, num_verts, num_edges);
+        c_extract_subgraph_vertex(in_graph_float, <int*>c_vertices, <int>num_input_vertices, out_graph_float);
+
+        tmp = rmm.device_array_from_ptr(<uintptr_t>out_graph_float.src_indices,
+                                        nelem=out_graph_float.number_of_edges,
+                                        dtype=np.int32)
+        df['src'] = cudf.Series(tmp)
+
+        tmp = rmm.device_array_from_ptr(<uintptr_t>out_graph_float.dst_indices,
+                                        nelem=out_graph_float.number_of_edges,
+                                        dtype=np.int32)
+
+        df['dst'] = cudf.Series(tmp)
+        if weights is not None:
+            tmp = rmm.device_array_from_ptr(<uintptr_t>out_graph_float.edge_data,
```

Comment:
I remember there were discussions on addressing https://github.com/rapidsai/cugraph/issues/572 along with other subgraph extraction improvements.
I can't find it in the PR. Am I missing something (if so this PR should close 572) or will this be addressed separately later on?

Reply:
I am going to construct a test for #572 and test it against this.  If it doesn't work I was planning on adding a fix to this PR.  Once I have that test in place I'll add the appropriate annotation.

Reply:
Added a fix to the PR down a few lines (after the comment: `renumber vertices to match original input`) TO ADDRESS #572.  Also updated the unit test to remove the [now] unnecessary lookup of ids in the vertex array.